oauthlib/oauth2/rfc6749/request_validator.py

```diff
@@ -166,6 +166,26 @@ def is_within_original_scope(self, request_scopes, refresh_token, request, *args
         """
         return False
 
+    def introspect_token(self, token, token_type_hint, request, *args, **kwargs):
+        """Introspect an access or refresh token.
+
+        Called once introspect token request is validated. This method
+        should return a dictionary with any desired claims associated
+        with the *token*. The implementation can use *token_type_hint*
+        to lookup this type first, but then it must fallback to other
+        types known, to be compliant with RFC.
+
+        The dict of claims is added to request.token after this method.
+
+        :param token: The token string.
+        :param token_type_hint: access_token or refresh_token.
+        :param request: The HTTP Request (oauthlib.common.Request)
+
+        Method is used by:
+            - Introspect Endpoint (all grants are compatible)
+        """
```

Comment:
Add a link to https://tools.ietf.org/html/rfc7662#section-2.2 here, and perhaps even mention the (currently) registered claims?

Reply:
Good idea !